dlib/data_io/cifar.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2020  Davis E. King (davis@dlib.net)
+// License: Boost Software License   See LICENSE.txt for the full license.
+#ifndef DLIB_CIFAR_CPp_
+#define DLIB_CIFAR_CPp_
+
+#include "cifar.h"
+#include <fstream>
+
+// ----------------------------------------------------------------------------------------
+
+namespace dlib
+{
+    void load_cifar_10_dataset (
+        const std::string& folder_name,
+        std::vector<matrix<rgb_pixel>>& training_images,
+        std::vector<unsigned long>& training_labels,
+        std::vector<matrix<rgb_pixel>>& testing_images,
+        std::vector<unsigned long>& testing_labels
+    )
+    {
+        using namespace std;
+        std::vector<string> file_names{
+            "data_batch_1.bin",
+            "data_batch_2.bin",
+            "data_batch_3.bin",
+            "data_batch_4.bin",
+            "data_batch_5.bin",
+            "test_batch.bin"
+        };
+        std::vector<ifstream> fins;
```

Comment:
It's kinda funny to make an array of ifstreams.  You should instead make them as you need them.  E.g.
```
for (const string filename : {"file1.bin", "file2.bin"})
{
    ifstream fin(folder_name + "/" + filename), ios::binary);
    ...
} 
```

Reply:
Yes, I was not totally happy about it either, but it made the code so much shorter... 😅